akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -605,6 +605,14 @@ import scala.util.control.NonFatal
     // call has no effect and therefore does the right thing: nothing.
     try {
       inputs.foreach(_.onInternalError(reason))
+      // fail all sources
+      logics.foreach { logic ⇒
+        // source
+        if (logic.inCount == 0 && logic.outCount > 0) {
```

Comment:
Only failing sources as they will propagate the failure downstream, if I just go through the logics it happens to the stages in random order (not quite sure if that matters, it just seemed more tidy and according to the existing logic, which only failed cross-boundary inputs).